qiskit/utils/deprecation.py

```diff
@@ -113,3 +113,117 @@ def _rename_kwargs(
                 )
 
                 kwargs[new_arg] = kwargs.pop(old_arg)
+
+
+# We insert deprecations in-between the description and Napoleon's meta sections. The below is from
+# https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#docstring-sections. We use
+# lowercase because Napoleon is case-insensitive.
+_NAPOLEON_META_LINES = frozenset(
+    {
+        "args:",
+        "arguments:",
+        "attention:",
+        "attributes:",
+        "caution:",
+        "danger:",
+        "error:",
+        "example:",
+        "examples:",
+        "hint:",
+        "important:",
+        "keyword args:",
+        "keyword arguments:",
+        "note:",
+        "notes:",
+        "other parameters:",
+        "parameters:",
+        "return:",
+        "returns:",
+        "raises:",
+        "references:",
+        "see also:",
+        "tip:",
+        "todo:",
+        "warning:",
+        "warnings:",
+        "warn:",
+        "warns:",
+        "yield:",
+        "yields:",
+    }
+)
+
+
+def add_deprecation_to_docstring(
+    func: Callable, msg: str, *, since: Optional[str], pending: bool
```

Comment:
what's the point of the `*`?

Reply:
It requires you to use keyword arguments for everything after the star. The reasons for that:

1. Call sites are clearer, rather than seeing a `True` or `False` with no context. IDEs make this less painful by showing what argument names are, but there are many contexts you read code w/o that, like GitHub or Vim.
2. More flexibility for the future of the API because we no longer need to worry about the order of positional arguments. Deprecating positional arguments is a pain.